### ISSUE TITLE
The ChromeCast feature with react-native-google-cast on Android (Expo) displays "noDevicesAvailable" when attempting to cast.

### ISSUE THREAD

**Issue by Ashish00100** [react-native-google-cast/react-native-google-cast#545]
I'm using react-native-google-cast in my React Native project for casting on both iOS and Android. While the casting feature is integrated into iOS and the casting button is visible, I encounter an issue where no devices are available when I click the button. However, the same setup works perfectly on Android, and the device is available there.

I have try the react-native-google-cast documentation but not able to resolve

**Comment by ChadyAyoub4** [comment]
@Ashish00100 Encoutering the same issue, are you using Xcode 16 to build your ios app?

**Comment by Ashish00100** [comment]
@ChadyAyoub4 No, I'm using Xcode version 15.4.

**Comment by ChadyAyoub4** [comment]
@Ashish00100  is the target device ios 18 maybe?

**Comment by Ashish00100** [comment]
@ChadyAyoub4 No, my iOS version is 17.4.1.

**Comment by ChadyAyoub4** [comment]
weird, i only have these issues when using Xcode 16, and ios 18, or using google-cast package version 4.8+

**Comment by Ashish00100** [comment]
@ChadyAyoub4 yes, my google-cast package version is 4.8.0.

**Comment by ChadyAyoub4** [comment]
Try downgrading to cast 4.6.1 and manually set the cast framework version to 4.7.0 in podfile.lock and then try again

**Comment by petrbela** [comment]
There were some issue in Google Cast SDK 4.8.0 and 4.8.1. Are you trying with the latest version 4.8.3.? Also, are you testing with a device or a simulator?

**Comment by Ashish00100** [comment]
@petrbela Yes, I am using a real device, and I tested version 4.8.3 as well, and the same issue was present.

**Comment by ChadyAyoub4** [comment]
I tested all versions above 4.8, they do work on simulators but not consistently, and on real devices they never worked for me, i still have to rely on V 4.6.1 for ios as it is the most stable version

**Comment by Ashish00100** [comment]
@ChadyAyoub4 I have changed podfile.lock to 4.7.0 but by running from xcode it says 

Issue : The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.

But if run pod install

Issue : [!] CocoaPods could not find compatible versions for pod "google-cast-sdk-dynamic-xcframework-no-bluetooth":
  In snapshot (Podfile.lock):
    google-cast-sdk-dynamic-xcframework-no-bluetooth (= 4.7.0)

  In Podfile:
    react-native-google-cast (from `../../../node_modules/react-native-google-cast`) was resolved to 4.6.1, which depends on
      react-native-google-cast/NoBluetoothArm (= 4.6.1) was resolved to 4.6.1, which depends on
        google-cast-sdk-dynamic-xcframework-no-bluetooth
        
  ====> And If I make the version 4.7.0 for casting same issue occurs no device available

**Comment by ChadyAyoub4** [comment]
@Ashish00100 Here is the full steps to make it work. 
Set the package version to 4.6.1 then run yarn or npm i
then run pod install
then look for google-cast-sdk-dynamic-xcframework-no-bluetooth (4.8.0)
set 4.8.0 to 4.7.1 (my mistake i wrote it earlier as 4.7.0) then run pod install again. it should work fine after this

**Comment by Ashish00100** [comment]
@ChadyAyoub4, I've made the changes you suggested, but I'm still encountering the same issue—no device found. 
Do I need to update the iOS version or Xcode version as well? 
Currently, I’m using Xcode 15.4 and iOS 17.4.1.

**Comment by ChadyAyoub4** [comment]
@Ashish00100 Very weird. Usually this fixes it for me. Did you make sure that local network permission is on in the settings of your phone?

**Comment by Ashish00100** [comment]
@ChadyAyoub4 Yes I have given local network permission

**Comment by Ashish00100** [comment]
@ChadyAyoub4 Yes, my app has been granted permission to access the local network.

**Comment by vinnyA3** [comment]
If anyone is still running into this ... I needed to explicitly set the `startDiscoveryAfterFirstTapOnCastButton` to `true` when setting up the SDK options  (ref:  https://react-native-google-cast.github.io/docs/getting-started/setup)
